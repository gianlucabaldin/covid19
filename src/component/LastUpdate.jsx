import React from 'react';
import moment from 'moment';
import { Link, Typography } from '@material-ui/core';

// const LastUpdate = ({ date, href }) => {
const LastUpdate = ({ date, href }) => (
  <Typography style={{ fontStyle: 'italic' }}>
    Data provided by <Link href={href}>{href}</Link>, last update available:{' '}
    {moment(date, 'YYYY-MM-DD').format('DD/MM/YYYY')}
  </Typography>
);

export default LastUpdate;
