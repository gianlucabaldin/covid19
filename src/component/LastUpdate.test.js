import React from 'react';
import { shallow } from 'enzyme';
import LastUpdate from './LastUpdate';

describe('Last Update component', () => {
  let wrapper = shallow(<LastUpdate />);
  it('should renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render date and ref', () => {
    wrapper = shallow(<LastUpdate date="2020-04-20" href="www.google.com" />);
    expect(wrapper).toIncludeText(
      'Data provided by www.google.com, last update available: 20/04/2020',
    );
  });
});
