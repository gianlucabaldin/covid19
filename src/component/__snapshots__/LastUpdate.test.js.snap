// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Last Update component should renders properly 1`] = `
<WithStyles(ForwardRef(Typography))
  style={
    Object {
      "fontStyle": "italic",
    }
  }
>
  Data provided by 
  <WithStyles(ForwardRef(Link)) />
  , last update available:
   
  Invalid date
</WithStyles(ForwardRef(Typography))>
`;
